### ISSUE TITLE
Could we respect NPM_VERSION for customer use of NPM?

### ISSUE THREAD

**Issue by fool** [netlify/build-image#138]
At present it appears that we only "respect" NPM version in case:

- there is a package.json in the base dir (https://github.com/netlify/build-image/blob/master/run-build-functions.sh#L323)
- and don't run yarn (https://github.com/netlify/build-image/blob/master/run-build-functions.sh#L330)

Some customers would like to specify an NPM_VERSION flag and have it work when for instance they have a monorepo without a /package.json, and cd to another directory to use "npm run test" in a build shell script or makefile rather than as invoked from "installing automatically from package.json".

This is a rare use case, and a request rather than a demand, but at present I would like to track it to see how frequent it is, so it's marked needs-review until it bubbles up with some more requests and we can re-assess whether it is worth the effort to support directly.

Current workaround:  intentionally install the npm version you need before running:

```npm install npm@$NPM_VERSION -g && npm run test```

requested by: https://app.intercom.io/a/apps/q245f50x/inbox/inbox/unassigned/conversations/14414749736

**Comment by brycekahle** [comment]
If they use `base`, then our current behavior will work correctly.

**Comment by brycekahle** [comment]
If they manually `cd` to another directory, a lot of our automatic Node behavior will not work, such as caching `node_modules`.

**Comment by fool** [comment]
Indeed.  But in a monorepo, you cannot set base if you want to build two or more separate code bases.  I am aware it is a heavy lift and that's why I said we're just using this to track demand, not ask for change yet

**Comment by brycekahle** [comment]
Can't you use contexts?

**Comment by brycekahle** [comment]
Ah, it was unclear they wanted to run commands in two different folders during the same build. I think the "workaround" you posted is the best solution at the moment.

**Comment by rockonedege** [comment]
I was the customer in question. Sorry being late to the discussion. But Chinese New Year is _this_ long. 

Here's my story.

At first I had my static site completely build with some Python scripts generating some html files and selected root site name: {my name}.netlify.com

Then  I found more usage of this site, such as  /blog, /cv etc.  each subdirectory built with its own tech stack., say one with webpack+Vue,  the other Python+plain html.   

Rather than manage a few sites,  I prefer an umbrella root site  {my name}.netlify.com. So I had to build each sub-directory for deployment with my own script,  and leave the index.html in the root directory as a site map.

It seems pretty natural and flexible for me build each subdirectory with different technologies.

**Comment by fool** [comment]
We think this is affecting yarn too in monorepos, and are doing some tests to confirm.  Reported in customer cases:

- https://app.intercom.io/a/apps/q245f50x/inbox/inbox/unassigned/conversations/18572475790
- https://app.intercom.io/a/apps/q245f50x/inbox/inbox/unassigned/conversations/18991508943

**Comment by futuregerald** [comment]
Another example showing that we are using the wrong yarn version when building from cache in an npm script. 

Here is a test site:
https://app.netlify.com/sites/netlify-lerna-yarn-test/deploys

Build command used there is: 
`yarn --version && yarn run version`

All this does is print the yarn version from terminal and also from inside of an npm/yarn script, and these are the results:

![image](https://user-images.githubusercontent.com/23725114/47033091-f21f0a00-d141-11e8-972e-4fb2e263f562.png)

HOWEVER, when clearing cache, you get the intended result:

![image](https://user-images.githubusercontent.com/23725114/47033182-272b5c80-d142-11e8-834b-2074fc434658.png)
